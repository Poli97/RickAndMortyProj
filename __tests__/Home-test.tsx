import React from 'react';
import HomeScreen from '../src/screens/home/Home';

jest.useFakeTimers();

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {
  StackNavigationProp,
  StackNavigatorRoutes,
  StackRouteProp,
} from 'src/routes/stackRouteList';

const navigation = {
  navigate: jest.fn(),
} as unknown as StackNavigationProp<StackNavigatorRoutes.Home>;

const route = {} as StackRouteProp<StackNavigatorRoutes.Home>;

it('renders correctly', () => {
  const tree = renderer.create(
    <HomeScreen navigation={navigation} route={route} />,
  );

  expect(tree.toJSON()).toMatchSnapshot();
});
