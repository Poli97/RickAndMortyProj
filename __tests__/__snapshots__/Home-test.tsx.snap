// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "paddingTop": 10,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
      />
    }
    refreshing={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
